### PR TITLE
bitcoin: update to 26.0

### PR DESCRIPTION
**Pull request by fanquake**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks @fanquake the update itself looks fine to me. However, if possible the build dependency should be updated to use `python312` as that is the current default Python version in MacPorts. Also, you could remove the `use_parallel_build yes` line as that is the default so it's redundant.

**Comment by pmetzger** [comment]
@fanquake If you fix the things @reneeotten has asked for, we can merge this.

**Comment by fanquake** [comment]
@pmetzger Sorry, this will be updated to include the suggestion tomorrow.

**Comment by fanquake** [comment]
> should be updated to use python312
> remove the use_parallel_build yes line

Updated to include these changes.

File: finance/bitcoin/Portfile [hunk 1]
```diff
@@ -26,22 +26,20 @@ long_description        Bitcoin is a peer-to-peer digital currency. By peer-to-p
...
+checksums               rmd160  b3b0060c79474128a27c4b6f4049e398497a63b5 \
+                        sha256  ab1d99276e28db62d1d9f3901e85ac358d7f1ebcb942d348a9c4e46f0fcdc0a1 \
+                        size    12754015
 
 depends_build           port:autoconf \
                         port:automake \
                         port:libtool \
                         port:pkgconfig \
-                        port:python311
+                        port:python312
```
Comment (ryandesign) [inline finance/bitcoin/Portfile:37]: You can't just change the dependency. You also have to tell the build to use this version of python. Looking through the history of the port, back to when the python dependency was originally added in #1268, I don't see how this has ever worked.

Comment (ryandesign) [reply finance/bitcoin/Portfile:37]: By which I mean, yes, the port builds, but I see no indication in the log that python is searched for or found. So either this software doesn't use python, in which case remove the dependency, or it searches for python by various names and uses whichever one it found, in which case fix it to use specifically this python that you've declared a dependency on.

**Comment by ryandesign** [comment]
> ###### Tested on
> macOS x.y Xcode x.y / Command Line Tools x.y.z

> * [ ]  tried a full install with `sudo port -vst install`?
> * [ ]  tested basic functionality of all binary files?

You didn't test this? You should always verify that a port builds and works before submitting a PR.

File: finance/bitcoin/Portfile (hunk 1 above)
Comment (fanquake) [reply finance/bitcoin/Portfile:37]: > but I see no indication in the log that python is searched for or found

The indication in the log is right here:
> checking for python3.12... /opt/local/bin/python3.12

**Comment by fanquake** [comment]
> You didn't test this? You should always verify that a port builds and works before submitting a PR.

I don't use macports. I'm submitting these PRs, because the currently maintainer clearly isn't maintaining the port, and submitting updates seems better than leaving the port to rot (or should ports with no maintainer be removed?). The multiple CIs should provide fairly good coverage that this builds? Outside of that, I'm fairly confident that the software works when compiled this way on macOS, because I am one of the upstream maintainers.

**Comment by pmetzger** [comment]
I'm inclined to commit this for now and perhaps improve it after. @ryandesign thoughts?

**Pull request merged**